CMakeLists.txt

```diff
@@ -141,7 +141,13 @@ else()
   add_cxx_compiler_flag(-fstrict-aliasing)
   # Disable warnings regarding deprecated parts of the library while building
   # and testing those parts of the library.
-  add_cxx_compiler_flag(-Wno-deprecated-declarations)
+  if (CMAKE_CXX_COMPILER_ID STREQUAL "Intel")
```

Comment:
I _think_ you can just try adding the flag. AddCXXCompilerFlag.cmake will check if the flag exists before adding it.

Or, as per the c++11:

```
add_cxx_compiler_flag(-Wno-deprecated-declarations)
if (NOT HAVE_CXX_FLAG_WNO_DEPRECATED_DECLARATIONS)
  add_cxx_compiler_flag(-wd1786)
endif()
```

maybe. it depends if the intel compiler supports the flag but ignores it, or returns an error if you try to set it.

Reply:
@LebedevRI yep, their frontend can get quite weird. A probably safer choice would be to tell the compiler to treat unrecognized flags as errors:

https://godbolt.org/g/SFSuFb

Reply:
Point was, it looks like there *is* a way to make it complain.
So sounds like you just need to fix `add_cxx_compiler_flag()`.

Reply:
@LebedevRI I was trying to fix `add_cxx_compiler_flag()` for `icpc` but this looks like an Intel regression to me:

https://godbolt.org/g/tfxWvW

If you're using `icc >= 17.0.4` (the `17.0.0` godbolt is using still works), the `-Wno-deprecated-declarations` option is considered recognized (`-diag-error=10006` is happy) but still ignored.

Reply:
@dominichamon I like obvious and explicit :+1: 

Ok, I will add that change and I'm going to try to open a call to Intel to fix the regression, let's just hope for a better future :)